bip32.c

```diff
@@ -94,7 +99,7 @@ int hdnode_from_xprv(uint32_t depth, uint32_t child_num, const uint8_t *chain_co
 	const curve_info *info = get_curve_by_name(curve);
 	if (info == 0) {
 		failed = true;
-	} else {
+	} else if (info->params) {
```

Comment:
This fix should probably go to master.  Can you put it in a separate pull request?

Currently this function is only called with "secp256k1", so it is not be exploitable, right?